### PR TITLE
Types: lib/url

### PR DESCRIPTION
**Pull request by sirreal**
Type lib/url

### CODE REVIEW

File: client/lib/url/index.ts [hunk 1]
```diff
@@ -60,19 +61,22 @@ export function isExternal( url ) {
...
- * @param  {String}  url The URL to remove http(s) from
- * @return {?String}     URL without the initial http(s)
+ * @param  url The URL to remove http(s) from
+ * @return     URL without the initial http(s)
  */
-export function withoutHttp( url ) {
+export function withoutHttp( url: '' ): '';
+export function withoutHttp( url: Falsey ): null;
+export function withoutHttp( url: URL ): URL;
+export function withoutHttp( url: URL | Falsey ): URL | null {
```
Comment (ockham) [inline client/lib/url/index.ts]: TIL how to specify fine-grained return types based on function signature :+1:

File: client/lib/url/index.ts [hunk 2]
```diff
@@ -186,10 +194,10 @@ export function omitUrlParams( url, paramsToOmit ) {
 /**
  * Wrap decodeURI in a try / catch block to prevent `URIError` on invalid input
  * Passing a non-string value will return an empty string.
- * @param  {String} encodedURI URI to attempt to decode
- * @return {String}            Decoded URI (or passed in value on error)
+ * @param  encodedURI URI to attempt to decode
+ * @return            Decoded URI (or passed in value on error)
  */
-export function decodeURIIfValid( encodedURI ) {
+export function decodeURIIfValid( encodedURI: string ): string {
```
Comment (ockham) [inline client/lib/url/index.ts]: Shouldn't the return type be `URL`?

Comment (sirreal) [reply client/lib/url/index.ts]: Seems reasonable 👍 

```suggestion
export function decodeURIIfValid( encodedURI: string ): URL {
```

**Comment by ockham** [comment]
The e2 canary is failing :roll_eyes: Might wanna try rebasing, that's what fixed it for me for #34624.

**Pull request merged**